x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -93,19 +96,30 @@ export const createLogThresholdExecutor = (libs: InfraBackendLibs) =>
     LogThresholdAlertState,
     LogThresholdAlertContext,
     LogThresholdActionGroups
-  >(async ({ services, params }) => {
-    const { alertWithLifecycle, savedObjectsClient, scopedClusterClient } = services;
-    const { sources } = libs;
-    const alertFactory: LogThresholdAlertFactory = (id, reason, value, threshold) =>
-      alertWithLifecycle({
+  >(async ({ services, params, startedAt }) => {
+    const { alertWithLifecycle, savedObjectsClient, scopedClusterClient, getAlertStartedDate } =
+      services;
+    const { sources, basePath } = libs;
+    let indexedStartedAt: string = startedAt.toISOString();
+    const alertFactory: LogThresholdAlertFactory = (id, reason, value, threshold) => {
+      indexedStartedAt = getAlertStartedDate(id) ?? indexedStartedAt;
+      return alertWithLifecycle({
         id,
         fields: {
           [ALERT_EVALUATION_THRESHOLD]: threshold,
           [ALERT_EVALUATION_VALUE]: value,
           [ALERT_REASON]: reason,
         },
       });
+    };
 
+    const relativeViewInAppUrl = getLogsAppAlertUrl(new Date(indexedStartedAt).getTime());
```

Comment:
Since this is a mutable variable modified via the closure of the `alertFactory` function it's hard to follow the dataflow as a reader. What happens if the `alertFactory` is called multiple times? Looks like the last call will win - is that intended?

And independent from the technical data flow can you help me understand why we want to use the alert-as-data start timestamp instead of the one that this rule execution is based on? Is it intentional that the timestamp doesn't update when successive executions cause the alert to stay active?

Reply:
@weltenwort, you are right, and this is intended. And actually, we are using the rule execution start timestamp. The fact is the rule registry took the **first** timestamp coming from the executor and indexed it (alert-as-data). But as the executor function is run before the indexing process in the rule registry, we pick up the start timestamp (in the first run) and use it. 
The workflow is:
- Get the `startedAt` timestamp from the executor function.
- Check via the rule registry service function `getAlertStartedDate`, If there is a value we pick it up (alert-as-data), if there is no value, it means this is the first run of the executor function and we use its timestamp (which will be the indexed timestamp)

Reply:
Maybe I misunderstand the control flow here, but how is the `alertFactory` supposed be be called between the initialization of `indexedStartedAt` and this line? Won't this always be `startedAt.toISOString()`?

Reply:
Done